site/src/pages/docs/server-graphql.mdx

```diff
@@ -0,0 +1,122 @@
+# Running a GraphQL service
+
+<Tip>
+
+Visit [armeria-examples](https://github.com/line/armeria-examples) to find a fully working example.
+
+</Tip>
+
+First, You need the `armeria-graphql` dependency:
+
+<RequiredDependencies
+  boms={[{ groupId: 'com.linecorp.armeria', artifactId: 'armeria-bom' }]}
+  dependencies={[
+    { groupId: 'com.linecorp.armeria', artifactId: 'armeria-graphql' },
+  ]}
+/>
+
+<Tip>
+
+Please note that this is entirely based on [graphql-java](https://www.graphql-java.com/).
+
+</Tip>
+
+- [GraphQL Java](https://www.graphql-java.com/) is the Java (server) implementation for GraphQL.
+- This documentation shows how to serve a service made from GraphQL Java over HTTP.
+- The main steps for creating a GraphQL service are:
+
+## Defining schema
+
+A schema which defines each field that can be queried or mutated and what types those fields are.
+This schema defines one top level field (in the type Query): which returns the details of a User.
+
+```graphql
```

Comment:
How about?

Let's assume we have the following [GraphQL Schema](https://graphql.org/learn/schema/):

```graphql
type Query {
  user(id: ID): User
}

type User {
  id: ID
  name: String
}
```

The schema describes that if a client requests a `User` data using the `ID` field, the GraphQL server will
return the `User` data. The GraphQL engine uses a [DataFetcher](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/DataFetcher.html)
to resolve the `ID` and fetch the corresponding object. For the sake of simplicity, let's implement
`UserDataFetcher` that stores `User` data in a `Map`:

```java
import graphql.schema.DataFetcher;
import graphql.schema.DataFetchingEnvironment;

class UserDataFetcher implements DataFetcher<User> {

    private final Map<String, User> data = Map.of("1", new User("1", "hero"),
                                                  "2", new User("2", "human"),
                                                  "3", new User("3", "droid"));

    @Override
    public User get(DataFetchingEnvironment environment) throws Exception {
        final String id = environment.getArgument("id");
        return data.get(id);
    }
}
```

## `GraphqlService`
Once you've finished the implementation of the `DataFetcher`, you need to build a <type://GraphqlService> using
a <type://GraphqlServiceBuilder> and add it to the <type://ServerBuilder>:

```java
import com.linecorp.armeria.server.Server;
import com.linecorp.armeria.server.ServerBuilder;

ServerBuilder sb = Server.builder();
...
sb.service("/graphql",
           GraphqlService.builder()
                         .runtimeWiring(c -> {
                            c.type("Query",
                                   typeWiring -> typeWiring.dataFetcher("user", new UserDataFetcher()));
                         })
                         .build());
...
Server server = sb.build();
server.start();
```

// explanation about runtimeWiring

## Blocking service implementation

...

Reply:
@freevie you might want to take a look at this PR. 😄

Reply:
I'm sorry for being late. 😅 I fixed